src/syntax/syntax.rkt

```diff
@@ -61,8 +69,9 @@
 ;; TODO: the costs below seem likely to be incorrect, and also do we still need them?
 (define-table operators
   [args  (listof (or/c '* natural-number/c))]
-  [bf    (unconstrained-argument-number-> (or/c bigfloat? boolean? bigcomplex?) (or/c bigfloat? boolean? bigcomplex?))]
-  [fl    (unconstrained-argument-number-> (or/c flonum? boolean? complex?) (or/c flonum? boolean? complex?))]
+  [bf    (unconstrained-argument-number-> (or/c bigfloat? boolean?) (or/c bigfloat? boolean? bigcomplex?))]
+  [fl    (unconstrained-argument-number-> (or/c flonum? boolean?) (or/c flonum? boolean? complex?))]
+  [nonffi (unconstrained-argument-number-> (or/c bigfloat? flonum? boolean?) (or/c bigfloat? flonum? boolean? complex? bigcomplex?))]
```

Comment:
Why do we allow the `nonffi` function to take `bigfloat?` input, or return `bigfloat?` output?